raiden/blockchain/filters.py

```diff
@@ -273,4 +294,8 @@ def union(self, other: "RaidenContractFilter") -> "RaidenContractFilter":
                 self.ignore_secret_registry_until_channel_found
                 and other.ignore_secret_registry_until_channel_found
             ),
+            # We must not have two different non-None service registries. Choose the non-None one.
```

Comment:
in which case happens this?

Reply:
In case something goes wrong. I don't think it can currently happen, but a similar logic was already present for the `default_registry` and I would hope this is a robust `union` logic for the future.